ql/src/java/org/apache/hadoop/hive/ql/exec/vector/expressions/aggregates/VectorUDAFBloomFilterMerge.java

```diff
@@ -602,4 +602,11 @@ public void assignRowColumn(VectorizedRowBatch batch, int batchIndex, int column
     Aggregation bfAgg = (Aggregation) agg;
     outputColVector.setVal(batchIndex, bfAgg.bfBytes, 0, bfAgg.bfBytes.length);
   }
+
+  /**
+   * Let's clone the batch when we're working in parallel, see HIVE-26655.
+   */
+  public boolean batchNeedsClone() {
+    return numThreads > 0;
+  }
```

Comment:
just a quick pass, if thread count is 1, you still need a clone? for parallel execution it should be more than 1 thread? or some catch?

Reply:
still need yes, thread=1 means the executor start processing the bloomfilter on 1 thread async while the main thread is fetching the next one